### ISSUE TITLE
test_compat flaky with randomized hashes

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5746]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)| @pitrou reported|
|-|-|
|Trac ID|trac#5746|
|Type|defect|
|Created|2012-07-08 20:53:49Z|

One test depends on hash ordering and fails sporadically with ``python -R``:

```
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/home/antoine/twisted/twisted/test/test_compat.py", line 186, in test_frozenset
    self.assertEqual(list(a), ['a', 'b'])
  File "/home/antoine/twisted/twisted/trial/unittest.py", line 270, in assertEqual
    % (msg, pformat(first), pformat(second)))
twisted.trial.unittest.FailTest: not equal:
a = ['b', 'a']
b = ['a', 'b']
```

Attaching patch.

Attachments:

* [randomhash_test_compat.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/06a/06a9faaae060b0ef27cabec6e75c775ff7a2c757/e61cec48b5c072ac78a7908c9f88ffaf60663d48.patch) (467 bytes) - added by antoine on 2012-07-08 20:53:55Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5746 5746
type__defect defect
reporter__antoine antoine
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1341780829000000 1341780829000000
changetime__1474197555542286 1474197555542286
version__None None
owner__ 
cc__antoine
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [34706]) Remove the assumption about set iteration order from the unit test for the frozenset compatibility api.

Author: antoine
Reviewer: exarkun
Fixes: [#5746](https://github.com/twisted/twisted/issues/5746)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
